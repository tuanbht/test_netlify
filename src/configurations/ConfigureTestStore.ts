import { configureStore, EnhancedStore } from '@reduxjs/toolkit';
import MockAdapter from 'axios-mock-adapter';
import axiosMiddleware from 'redux-axios-middleware';
import RootReducer from '../reducers';
import axiosClient from './ApiClient';

export const testStore = (initialState = {}): EnhancedStore =>
  configureStore({
    preloadedState: initialState,
    reducer: RootReducer,
    middleware: [axiosMiddleware(axiosClient)],
  });

export const mockAxios = new MockAdapter(axiosClient);
