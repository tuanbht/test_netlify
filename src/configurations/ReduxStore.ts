import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import axiosMiddleware from 'redux-axios-middleware';
import axiosClient from './ApiClient';
import rootReducer from '../reducers';
import logger from 'redux-logger';

const history = createBrowserHistory();

const middleware = [
  axiosMiddleware(axiosClient),
  ...getDefaultMiddleware({
    serializableCheck: false,
  }),
  logger,
];

const store = configureStore({
  reducer: rootReducer,
  middleware: middleware,
});

export { history, store };
