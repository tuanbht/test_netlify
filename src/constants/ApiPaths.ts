export const API_URL = process.env.REACT_APP_BASE_API;

export const ORDER_DETAILS_PATH = '/orders/:id/payment/';
export const MARK_AS_PAID_PATH = '/orders/:id/mask_as_paid/';
export const CANCEL_ORDER_PATH = '/orders/:id/payment/cancel/';
