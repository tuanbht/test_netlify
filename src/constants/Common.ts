export const VIEW_CRYPTO_TRANSACTION = process.env.REACT_APP_VIEW_CRYPTO_TRANSACTION;
