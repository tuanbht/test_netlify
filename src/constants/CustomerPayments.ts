import React from 'react';
import { ReactComponent as IconBTC } from '../assets/images/icons/icon-btc.svg';
import { ReactComponent as IconETH } from '../assets/images/icons/icon-eth.svg';
import { ReactComponent as IconUSDT } from '../assets/images/icons/icon-usdt.svg';

export const STEPS = {
  CRYPTO: 0,
  WALLET: 1,
  PROCESSING: 2,
  COMPLETE: 3,
};

export interface CryptoCurrencies {
  prefix: string;
  shortName: string;
  fullName: string;
  logo: React.FC<React.SVGProps<SVGSVGElement>>;
  walletAddress: string;
  amount: number;

  setWalletAddress(walletAddress: string): void;
  setAmount(amount: number): void;
}

export class CryptoCurrencies implements CryptoCurrencies {
  static initWithData(
    prefix: string,
    shortName: string,
    fullName: string,
    logo: React.FC<React.SVGProps<SVGSVGElement>>,
  ): CryptoCurrencies {
    const crypto = new CryptoCurrencies();
    crypto.prefix = prefix;
    crypto.shortName = shortName;
    crypto.fullName = fullName;
    crypto.logo = logo;
    return crypto;
  }
  setWalletAddress(walletAddress: string): void {
    this.walletAddress = walletAddress;
  }
  setAmount(amount: number): void {
    this.amount = amount;
  }
}

export const CRYPTO_CURRENCIES = {
  BITCOIN: CryptoCurrencies.initWithData('bitcoin', 'BTC', 'Bitcoin', IconBTC),
  ETHEREUM: CryptoCurrencies.initWithData('ethereum', 'ETH', 'Ethereum', IconETH),
  USDT: CryptoCurrencies.initWithData('usdt', 'USDT', 'USD Tether', IconUSDT),
};
