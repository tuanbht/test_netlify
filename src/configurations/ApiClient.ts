import Axios from 'axios';
import { API_URL } from 'constants/ApiPaths';
import { NOT_FOUND_PATH } from 'constants/RouterPaths';
import { store } from './ReduxStore';

const axiosClient = Axios.create({
  baseURL: API_URL,
  headers: {
    Accept: 'application/vnd.api+json',
    'Content-Type': 'application/vnd.api+json',
  },
});

axiosClient.interceptors.request.use((config) => {
  const token = store.getState().Credential.token;
  if (token) config.headers['Payment-Token'] = token;
  else window.location.assign(NOT_FOUND_PATH);
  return config;
});

export default axiosClient;
