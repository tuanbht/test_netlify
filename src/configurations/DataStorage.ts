import { ORDER_ID, PAYMENT_TOKEN } from '../constants/StoreKeys';

const setToken = (token: string): void => {
  localStorage.setItem(PAYMENT_TOKEN, token);
};

const setOrderId = (orderId: number): void => {
  localStorage.setItem(ORDER_ID, String(orderId));
};

const getToken = (): string | null => {
  return localStorage.getItem(PAYMENT_TOKEN);
};

const getOrderId = (): number => {
  return parseInt(localStorage.getItem(ORDER_ID) as string);
};

const clearAll = (): void => {
  localStorage.clear();
};

export default {
  setOrderId,
  setToken,
  getToken,
  getOrderId,
  clearAll,
};
