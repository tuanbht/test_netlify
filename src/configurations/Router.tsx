import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { NOT_FOUND_PATH, ORDER_CANCELLED_PATH, ORDER_EXPIRED_PATH, ORDER_PATH, ROOT_PATH } from 'constants/RouterPaths';
import { ORDER_CANCELLED_CONTENT, ORDER_EXPIRED_CONTENT } from 'constants/NotificationContent';
import { CustomerPayments, NotFound, NotificationPage, HomePage } from 'containers';

export const Router = (): React.ReactElement => {
  return (
    <Switch>
      <Route exact path={ROOT_PATH} component={HomePage} />
      <Route exact path={ORDER_CANCELLED_PATH} render={() => <NotificationPage content={ORDER_CANCELLED_CONTENT} />} />
      <Route exact path={ORDER_EXPIRED_PATH} render={() => <NotificationPage content={ORDER_EXPIRED_CONTENT} />} />
      <Route exact path={ORDER_PATH} component={CustomerPayments} />
      <Route exact path={NOT_FOUND_PATH} component={NotFound} />
      <Route render={() => <Redirect to={NOT_FOUND_PATH} />} />
    </Switch>
  );
};

export default Router;
