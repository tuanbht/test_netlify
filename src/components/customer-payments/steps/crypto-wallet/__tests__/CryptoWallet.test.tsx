import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import CryptoWallet from '../index';
import { CryptoCurrencies } from '../../../../../constants/CustomerPayments';
import { Button, Modal } from '@material-ui/core';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { testStore } from '../../../../../configurations/ConfigureTestStore';

jest.mock('copy-to-clipboard');

describe('CryptoWallet', () => {
  const CryptoLogo = () => React.createElement('svg');
  const crypto = CryptoCurrencies.initWithData('crypto-prefix', 'crypto-short-name', 'crypto-full-name', CryptoLogo);
  crypto.setWalletAddress('wallet-address');
  crypto.setAmount(0);
  const nextStep = jest.fn();
  let container: ReactWrapper;

  beforeEach(() => {
    jest.spyOn(global, 'setTimeout');
    HTMLCanvasElement.prototype.getContext = jest.fn();

    container = mount(
      <Provider store={testStore()}>
        <CryptoWallet crypto={crypto} nextStep={nextStep} />
      </Provider>,
    );
  });

  it('renders template correctly', () => {
    expect(container).toMatchSnapshot();
  });

  describe('click qr code button', () => {
    beforeEach(() => {
      container.find(Button).find({ children: 'QR Code' }).first().simulate('click');
    });

    it('should open qr code modal', () => {
      expect(container.find(Modal).props().open).toBe(true);
    });
  });

  describe('click outside of qr code modal', () => {
    beforeEach(() => {
      container.find(Button).find({ children: 'QR Code' }).first().simulate('click');
      expect(container.find(Modal).props().open).toBe(true);
      act(() => {
        container.find(Modal).props().onClose?.({}, 'backdropClick');
      });
    });

    it('should close qr code modal', async () => {
      container.update();
      expect(container.find(Modal).props().open).toBe(false);
    });
  });

  describe('click mark as paid button', () => {
    beforeEach(() => {
      container.find(Button).find({ children: 'Mark as Paid' }).first().simulate('click');
    });

    it('should call next step', () => {
      expect(nextStep).toBeCalled();
    });
  });

  describe('copy to clipboard', () => {
    describe('copy amount value', () => {
      const copyButton = (): ReactWrapper => (container as ReactWrapper).find(Button).find('#copy-amount').first();

      beforeEach(() => {
        jest.useFakeTimers();
        copyButton().simulate('click');
      });

      it('changes text of button to copied', () => {
        container.update();
        expect((copyButton().props() as { children: string }).children).toBe('Copied!');
      });

      it('will change to copy after 5 seconds', () => {
        act(() => {
          jest.runAllTimers();
        });
        expect(setTimeout).toBeCalled();
        container.update();
        expect((copyButton().props() as { children: string }).children).toBe('Copy');
      });

      describe('click on copied button', () => {
        beforeEach(() => {
          copyButton().simulate('click');
        });

        it('should not change any thing', () => {
          expect((copyButton().props() as { children: string }).children).toBe('Copied!');
        });
      });
    });

    describe('copy wallet address value', () => {
      const copyButton = (): ReactWrapper => (container as ReactWrapper).find(Button).find('#copy-wallet').first();

      beforeEach(() => {
        jest.useFakeTimers();
        copyButton().simulate('click');
      });

      it('changes text of button to copied', () => {
        expect((copyButton().props() as { children: string }).children).toBe('Copied!');
      });

      it('will change to copy after 5 seconds', () => {
        act(() => {
          jest.runAllTimers();
        });
        container.update();
        expect(setTimeout).toBeCalled();
        expect((copyButton().props() as { children: string }).children).toBe('Copy');
      });
    });
  });
});
