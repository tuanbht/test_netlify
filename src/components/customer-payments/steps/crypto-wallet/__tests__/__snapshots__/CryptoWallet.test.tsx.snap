// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoWallet renders template correctly 1`] = `
<div>
  <div
    className="makeStyles-title-14"
  >
    To make a payment, send 
    crypto-short-name
     to the address below
  </div>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-walletInfo-4"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <div
        className="makeStyles-label-6"
      >
        <CryptoLogo />
        <span
          className="makeStyles-labelName-7"
        >
          crypto-full-name
           Amount
        </span>
        <span
          className="makeStyles-labelAmount-8"
        >
          (76.25 USD)
        </span>
      </div>
      <div
        className="makeStyles-content-9"
      >
        <WithStyles(ForwardRef(OutlinedInput))
          classes={
            Object {
              "input": "makeStyles-contentInfoText-11",
              "root": "makeStyles-contentInfo-10",
            }
          }
          endAdornment={
            <ForwardRef(WithStyles)
              className="makeStyles-contentInfoCryptoName-12"
              position="end"
            >
              crypto-short-name
            </ForwardRef(WithStyles)>
          }
          fullWidth={true}
          readOnly={true}
          value={0}
        />
        <WithStyles(ForwardRef(Button))
          className="makeStyles-root-17 makeStyles-root-19 makeStyles-copyButton-13"
          id="copy-amount"
          onClick={[Function]}
        >
          Copy
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-walletAddress-5"
      item={true}
    >
      <div
        className="makeStyles-label-6"
      >
        <CryptoLogo />
        <span
          className="makeStyles-labelName-7"
        >
          crypto-full-name
           Address
        </span>
      </div>
      <div
        className="makeStyles-content-9"
      >
        <WithStyles(ForwardRef(OutlinedInput))
          classes={
            Object {
              "input": "makeStyles-contentInfoText-11",
              "root": "makeStyles-contentInfo-10",
            }
          }
          fullWidth={true}
          readOnly={true}
          value="wallet-address"
        />
        <WithStyles(ForwardRef(Button))
          className="makeStyles-root-17 makeStyles-root-19 makeStyles-copyButton-13"
          id="copy-wallet"
          onClick={[Function]}
        >
          Copy
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <ForwardRef(Modal)
    className="makeStyles-qrCodeModal-1"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-qrCodeContent-2"
      elevation={3}
    >
      <div
        className="makeStyles-qrCodeLabel-3"
      >
        <p>
          0
        </p>
        <p>
          crypto-short-name
        </p>
      </div>
      <QRCode
        bgColor="#FFFFFF"
        fgColor="#000000"
        includeMargin={false}
        level="L"
        renderAs="canvas"
        size={128}
        value="crypto-prefix:wallet-address?amount=0"
      />
    </WithStyles(ForwardRef(Paper))>
  </ForwardRef(Modal)>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-qrCodeButton-15"
    onClick={[Function]}
  >
    QR Code
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-paidButton-16"
    onClick={[Function]}
  >
    Mark as Paid
  </WithStyles(ForwardRef(Button))>
</div>
`;
